apps/src/lib/proto/types.rs

```diff
@@ -0,0 +1,329 @@
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+
+use prost::Message;
+use thiserror::Error;
+
+use super::generated::{services, types};
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("Error decoding a transaction from bytes: {0}")]
+    TxDecodingError(prost::DecodeError),
+    #[error("Error decoding an IntentGossipMessage from bytes: {0}")]
+    IntentDecodingError(prost::DecodeError),
+    #[error("Error decoding an DkgGossipMessage from bytes: {0}")]
+    DkgDecodingError(prost::DecodeError),
+    #[error("Intent is empty")]
+    NoIntentError,
+    #[error("Dkg is empty")]
+    NoDkgError,
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Debug)]
+pub struct Tx {
+    inner: types::Tx,
+}
+
+impl Tx {
+    pub fn from(tx_bytes: impl AsRef<[u8]>) -> Result<Self> {
+        let inner = types::Tx::decode(tx_bytes.as_ref())
+            .map_err(Error::TxDecodingError)?;
+        Ok(Tx { inner })
+    }
```

Comment:
I'm actually not sure, but I think the classic way of implementing this trait is : 

```rust
impl From<Into<Vec<u8>>> for Result<Tx> {
      fn from(tx_bytes: impl AsRef<[u8]>) ->Result<Self> {
              let inner = types::Tx::decode(tx_bytes.as_ref())
                  .map_err(Error::TxDecodingError)?;
              Ok(Tx { inner })
}
impl Tx {
...
```

Reply:
It should be `TryFrom<&[u8]>` as `From` conversions must not fail

Reply:
Thank you for your suggestion! I used `TryFrom`.